android/src/main/java/id/animo/mdocdatatransfer/MdocDataTransfer.kt

```diff
@@ -21,52 +21,55 @@ class MdocDataTransfer(
-                }
-
-                is TransferEvent.Connecting -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Connecting")
-                }
-
-                is TransferEvent.Connected -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Connected")
-                }
-
-                is TransferEvent.Disconnected -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Disconnected")
-                }
-
-                is TransferEvent.Error -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Error")
-                }
-
-                is TransferEvent.Redirect -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Redirect")
-                }
-
-                is TransferEvent.RequestReceived -> {
-                    Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.RequestReceived")
-                    val request = event.request as DeviceRequest
-                    sendEvent(
-                        MdocDataTransferEvent.ON_REQUEST_RECEIVED, mapOf(
-                            "deviceRequest" to request.deviceRequestBytes.asList(),
-                            "sessionTranscript" to request.sessionTranscriptBytes.asList()
+        val transferEventListener =
+            TransferEvent.Listener { event ->
+                when (event) {
+                    is TransferEvent.QrEngagementReady -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.QrEngagementReady")
+                        onQrEngagementReady?.let { it(event.qrCode.content) }
+                    }
+
+                    is TransferEvent.Connecting -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Connecting")
+                    }
+
+                    is TransferEvent.Connected -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Connected")
+                    }
+
+                    is TransferEvent.Disconnected -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Disconnected")
+                    }
+
+                    is TransferEvent.Error -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Error")
+                    }
+
+                    is TransferEvent.Redirect -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.Redirect")
+                    }
+
+                    is TransferEvent.RequestReceived -> {
+                        Log.d(TAG, ":::mdoc-data-transfer::: TransferEvent.RequestReceived")
+                        val request = event.request as DeviceRequest
+                        sendEvent(
+                            MdocDataTransferEvent.ON_REQUEST_RECEIVED,
+                            mapOf(
+                                "deviceRequest" to request.deviceRequestBytes.joinToString(":"),
```

Comment:
can't we send bytes from JS <-> Android/Swift? In this case I rather encode it as Base64 or hex